util/ivmFactory.js

```diff
@@ -38,31 +38,64 @@ async function createIvm(code, libraryVersionIds) {
       const eventMessages = events.map(event => event.message);
       const eventsMetadata = {};
       events.forEach(ev => {
-        eventsMetadata[ev.message.messageId] = ev.metadata;
+        eventsMetadata[ev.metadata ? ev.metadata.messageId : ev.message.messageId] = ev.metadata;
       });
 
+      const isObject = (o) => Object.prototype.toString.call(o) === '[object Object]';
```

Comment:
Is this safe to use instead `o.constructor == Object`
Just because it looks better. Not important though

Reply:
`o.constructor` returns a function. comparing it would be difficult. this seems to be recommended by more in stackoverflow etc